src/nbl/asset/CShaderIntrospector.cpp

```diff
@@ -699,15 +699,15 @@ static void introspectStructType(spirv_cross::Compiler& _comp, impl::SShaderMemo
             }
         }
 		else if (mtype.basetype != spirv_cross::SPIRType::Struct) // might have to ignore a few more types than structs
-			member.arrayStride = core::max(0x1u<<core::findMSB(member.size),16u);
+			member.arrayStride = core::max(0x1u<<core::findMSB(member.size),16u); // @Crisspl  NOT TRUE FOR MATRICES!
 
         if (mtype.basetype == spirv_cross::SPIRType::Struct) //recursive introspection done in DFS manner (and without recursive calls)
             _pushStack.push({member.members, mtype, member.offset});
         else
 		{
             member.mtxRowCnt = mtype.vecsize;
             member.mtxColCnt = mtype.columns;
-            if (member.mtxColCnt > 1u)
+            if (member.mtxColCnt > 1u) // @Crisspl why are you checking columns to get stride? What about row major?
```

Comment:
im checking if the member is of matrix type

Reply:
why do you need to check to figure out the stride?

Reply:
because only matrices have matrix strides (distance between rows/colums)

Reply:
if its row major shouldn't you be checking `member.mtxRowCnt` instead of `member.mtxColCnt` or?

Reply:
no, why?
first of all, row_major is a memory layout qualifier and nothing more
second thing is here im just checking if im dealing with matrix type. To do its best to check columnCount>1 since only matrix types have more than 1 columns

Reply:
ok fine... just double checking